test/e2e/specs/editor/various/block-comments.spec.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	blockCommentUtils: async ( { page, editor }, use ) => {
+		await use( new BlockCommentUtils( { page, editor } ) );
+	},
+} );
+
+test.describe( 'Block Comments', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.setGutenbergExperiments( [
+			'gutenberg-block-comment',
+		] );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllComments( 'block_comment' ),
+			requestUtils.setGutenbergExperiments( [] ),
+		] );
+	} );
+
+	test( 'can pin and unpin comments sidebar', async ( {
+		page,
+		blockCommentUtils,
+	} ) => {
+		const topBarButton = await blockCommentUtils.openBlockCommentSidebar();
+		await page
+			.getByRole( 'button', { name: 'Unpin from toolbar' } )
+			.click();
+		await expect( topBarButton ).toBeHidden();
+		await page.getByRole( 'button', { name: 'Pin to toolbar' } ).click();
+		await expect( topBarButton ).toBeVisible();
+	} );
+
+	test( 'can add a comment to a block', async ( {
+		page,
+		blockCommentUtils,
+	} ) => {
+		await blockCommentUtils.addBlockWithComment( {
+			type: 'core/paragraph',
+			content: 'Testing block comments',
```

Comment:
```suggestion
			attributes: {
				content: 'Testing block comments',
			},
```

Would you like to change the parameters so that we can use this utility for other blocks in the future?